src/utils.ts

```diff
@@ -20,4 +21,27 @@ export const camelCaseKeys = (rawFlags: LDFlagSet) => {
   return flags;
 };
 
-export default { camelCaseKeys };
+/**
+ * Gets the flags to pass to the provider from the changeset.
+ *
+ * @param changes the `LDFlagChangeset` from the ldClient onchange handler.
+ * @param targetFlags if targetFlags are specified, changes to other flags are ignored and not returned in the
+ * flattened `LDFlagSet`
+ * @param reactOptions reactOptions.useCamelCaseFlagKeys is used to determine whether to
+ * @return an `LDFlagSet` with the current flag values from the LDFlagChangeset filtered by `targetFlags`
+ * or null if none of the targetFlags were changed
+ */
+export const getFlattenedFlagsFromChangeset = (changes: LDFlagChangeset, targetFlags: LDFlagSet | undefined,
+                                               reactOptions: LDReactOptions): LDFlagSet | null => {
+  const flattened: LDFlagSet = {};
+  for (const key in changes) {
+    if (targetFlags === undefined || targetFlags[key] !== undefined) {
+      // tslint:disable-next-line:no-unsafe-any
+      const flagKey = reactOptions.useCamelCaseFlagKeys ? camelCase(key) : key;
+      flattened[flagKey] = changes[key].current;
+    }
+  }
+  return Object.keys(flattened).length > 0 ? flattened : null;
```

Comment:
Yeah, in most cases I would agree as it's typically the case that an empty map or array returned from a function will be used in exactly the same way as a collection with members. But in this case the consumers are going to take a conditional action based on whether the map has keys or not. In this case I thought the contract was more clear by returning null instead of empty map. It also means the consumers don't have to do this Object.keys check themselves.

I'm happy to change it, but just wanted to explain my reasoning.

Reply:
I understand your reasoning. Part of my reasoning is that this simplifies the utility function and removes the conditional typing from the return type, all while only adding minimal complexity to the consumers. All in all I think what you have in your latest changeset is better, thanks.

Reply:
I understand your reasoning. Part of my reasoning is that this simplifies the utility function and removes the conditional typing from the return type, all while only adding minimal complexity to the consumers. All in all I think what you have in your latest changeset is better, thanks.